src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1505,6 +1608,49 @@ CreateHashDistributedTableShards(Oid relationId, int shardCount,
 }
 
 
+/*
+ * CreateHashDistributedTableShards creates the shard of given null-shard-key
+ * distributed table.
+ */
+static void
+CreateNullShardKeyDistTableShard(Oid relationId, Oid colocatedTableId,
+								 bool localTableEmpty)
+{
+	/*
+	 * Decide whether to use exclusive connections per placement or not. Note that
+	 * if the local table is not empty, we cannot use sequential mode since the COPY
+	 * operation that'd load the data into shards currently requires exclusive
+	 * connections.
+	 */
+	bool useExclusiveConnection = false;
+	if (RegularTable(relationId))
+	{
+		useExclusiveConnection = CanUseExclusiveConnections(relationId,
+															localTableEmpty);
+	}
```

Comment:
maybe we should always set useExclusiveConnection to false?

Reply:
(because we're anyway creating one shard)

Reply:
Yes, agree, this is a bit of a hack to force create_distributed_table to use a separate connection per shard, such that a subsequent data copy into the table also uses a connection per shard.